### PR TITLE
New class helpers for Search.

### PR DESCRIPTION
**Pull request by mickel1138**
Added SearchOptions and SearchResults.

Also JSONSerializer util.

### CODE REVIEW

File: src/Util/JsonSerializer.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    /**
+     * Creates a response from JSON.
+     *
+     * @return Response
+     */
+    public function createJsonResponse(): Response
+    {
+        $header = array('Content-Type', 'application/json');
+        return new Response($this->json, Response::HTTP_OK, $header);
```
Comment (praneethpr) [inline src/Util/JsonSerializer.php:74]: Could use `new JsonResponse($this->json, Response::HTTP_OK)`

File: assets/js/vue/components/info-search/ResultSet.vue [hunk 2]
```diff
@@ -14,7 +14,7 @@
             <!-- Need response data to get facet data -->
 <!--            <Facet :facet-info="results.facetInfo.productTypeDesc" :facet-name="facetLabels.productTypeDesc" v-on="$listeners" :formValues="formValues"/>-->
 <!--            <Facet :facet-info="results.facetInfo.digitalTypeDesc" :facet-name="facetLabels.digitalTypeDesc" v-on="$listeners" :formValues="formValues"/>-->
-<!--            <Facet :facet-info="results.facetInfo.researchGroupsInfo" :facet-name="facetLabels.researchGroup" v-on="$listeners" :formValues="formValues"/>-->
+           <Facet :facet-info="results.facetInfo.researchGroupsInfo" :facet-name="facetLabels.researchGroup" v-on="$listeners" :formValues="formValues"/>
```
Comment (praneethpr) [inline assets/js/vue/components/info-search/ResultSet.vue:17]: It is variable issue, the api was returning `researchGroupInfo`, made that change in my branch and some other required changes. Please undo this change and pull my branch, then we can uncomment this line and other facets too

File: src/Search/InformationProductSearch.php [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            $researchGroupQueryTerm->setTerms($searchOptions->getResearchGroupFilter());
+            $researchGroupNameQuery->setQuery($researchGroupQueryTerm);
+            $boolQuery->addFilter($researchGroupNameQuery);
+        }
+
+        $query = new Query();
+
+        $query->setQuery($boolQuery);
+
+        $researchGroupNestedAggregation = new AggregationNested('researchGroupsAgg', 'researchGroups');
```
Comment (praneethpr) [inline src/Search/InformationProductSearch.php:86]: Could we move each query to a separate function?

File: src/Search/SearchOptions.php [hunk 4]
```diff
@@ -0,0 +1,177 @@
+<?php
```
Comment (praneethpr) [inline src/Search/SearchOptions.php:1]: I really like the idea of having a Util class instead of a function in the Controller like what i did last time 👍

File: src/Search/SearchOptions.php [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+     * @var integer $currentPage
+     */
+    private $currentPage = 1;
+
+    /**
+     * The maximum results per page.
+     *
+     * @var integer $maxPerPage
+     */
+    private $maxPerPage = 1000;
```
Comment (praneethpr) [inline src/Search/SearchOptions.php:36]: Maybe this number could be a 100?

Comment (mickel1138) [reply src/Search/SearchOptions.php:36]: As soon as the paginator is added, this could be set back to default 10.

File: src/Search/InformationProductSearch.php [hunk 6]
```diff
@@ -70,6 +70,28 @@ public function search(SearchOptions $searchOptions): SearchResults
         $publishedQueryTerm->setTerm('published', $isPublished);
         $boolQuery->addFilter($publishedQueryTerm);
 
+        $this->addFilters($boolQuery, $searchOptions);
```
Comment (mickel1138) [inline src/Search/InformationProductSearch.php:73]: I guess the published filter above can be added to?

File: src/Search/InformationProductSearch.php [hunk 7]
```diff
@@ -70,6 +70,28 @@ public function search(SearchOptions $searchOptions): SearchResults
...
+
+        $query = new Query();
+        $query->setQuery($boolQuery);
```
Comment (mickel1138) [inline src/Search/InformationProductSearch.php:76]: This can be done in one shot using the constructor if we like.

File: src/Search/SearchOptions.php [hunk 8]
```diff
@@ -209,4 +222,32 @@ public function isProductTypeDescFilterSet(): bool
     {
         return isset($this->productTypeDescFilter);
     }
+
+    /**
+     * @return array
```
Comment (mickel1138) [inline src/Search/SearchOptions.php:227]: Needs function description line (first)

**Review by fightingtexasaggie — APPROVED** [review]
